HABApp/openhab/http_connection.py

```diff
@@ -320,6 +321,53 @@ async def async_get_item(self, item_name: str) -> dict:
         else:
             return await ret.json(encoding='utf-8')
 
+    async def async_thing_remove_link(self, thing_uid: str, channel: str, item_name: str) -> bool:
+        if self.config.general.listen_only:
+            return False
+
+        # rest/links/ endpoint needs the channel to be url encoded 
+        # (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
+        # otherwise the REST-api returns HTTP-Status 500 InternalServerError
+        channel_uid = urllib.parse.quote(f"{thing_uid}:{channel}")
+        url = self.__get_openhab_url('rest/links/{item:s}/{channel:s}', item=item_name, channel=channel_uid)
```

Comment:
Imho we should move quoting into the function

```python
    def __get_openhab_url(self, url: str, *args, **kwargs) -> str:
        assert not url.startswith('/')

        url = url.format(*args, **kwargs)
        if url.startswith('rest/links/'):
            # rest/links/ endpoint needs the channel to be url encoded 
            # (AAAA:BBBB:CCCC:0#NAME -> AAAA%3ABBBB%3ACCCC%3A0%23NAME)
            # otherwise the REST-api returns HTTP-Status 500 InternalServerError
            url = urllib.parse.quote(url)
        return f'http://{self.__host:s}:{self.__port:d}/{url:s}'
```

Reply:
done in d5a9486